libscript/src/module-arithmetic.cpp

```diff
@@ -473,9 +475,11 @@ extern "C" MC_DLLEXPORT_DEF MCStringRef MCArithmeticExecFormatNumberAsString(MCN
 extern "C" MC_DLLEXPORT_DEF MCValueRef MCArithmeticExecParseStringAsNumber(MCStringRef p_operand)
 {
     double t_converted;
-    if (!MCTypeConvertStringToReal(p_operand, t_converted))
+
+    if (!(MCStringGetNumericValue(p_operand, t_converted) ||
+          MCTypeConvertStringToReal(p_operand, t_converted)))
```

Comment:
Doesn't formulating it like this mean `MCTypeConvertStringToReal` is called even if `MCStringGetNumericValue` fails?